proxy/src/control/discovery.rs

```diff
@@ -516,11 +510,9 @@ impl<T> DestinationSet<T>
             match rx.poll() {
                 Ok(Async::Ready(Some(update))) => match update.update {
                     Some(PbUpdate2::Add(a_set)) => {
-                        let set_labels = Arc::new(a_set.metric_labels);
```

Comment:
This seems correct ---  the `Arc` was originally added when the labels were formatted at request time rather than when the `WeightedAddrSet` was received, so I wanted to avoid cloning a string for every `WeightedAddr` in the `WeightedAddrSet`, but when the formatting was made eager this became vestigial. Thanks for catching that!